### PR TITLE
Experimental/sf

### PR DESCRIPTION
**Pull request by lqdung-lockon**

### CODE REVIEW

File: EventSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            EccubeEvents::FRONT_PRODUCT_INDEX_SEARCH => ['onFrontProductIndexSearch', 10],
```
Comment (lqdung-lockon) [inline EventSubscriber.php]: You can use QueryKey::PRODUCT_SEARCH event for these two events

File: Repository/ProductRankRepository.php [hunk 2]
```diff
@@ -23,132 +23,155 @@
...
+use Doctrine\ORM\EntityManagerInterface;
 
 /**
  * ProductRank
  *
  * This class was generated by the Doctrine ORM. Add your own custom
  * repository methods below.
  */
-class ProductRankRepository extends EntityRepository
+class ProductRankRepository
```
Comment (lqdung-lockon) [inline Repository/ProductRankRepository.php:37]: Please extend `AbstractRepository`.

File: Resource/template/admin/product_rank.twig [hunk 3]
```diff
@@ -19,18 +19,18 @@ You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 #}
-{% extends 'default_frame.twig' %}
+{% extends '@admin/default_frame.twig' %}
```
Comment (lqdung-lockon) [inline Resource/template/admin/product_rank.twig]: Please change to new styleguide `{% extends '@admin/styleguide_frame.twig' %}`

File: Resource/template/admin/product_rank.twig [hunk 4]
```diff
@@ -19,18 +19,18 @@ You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 #}
-{% extends 'default_frame.twig' %}
+{% extends '@admin/default_frame.twig' %}
 
 {% set menus = ['product', 'class_category'] %}
 
 {% block title %}商品管理{% endblock %}
 {% block sub_title %}商品並び替え{% endblock %}
```
Comment (lqdung-lockon) [inline Resource/template/admin/product_rank.twig]: Please change to message translator

**Comment by lqdung-lockon** [comment]
image

- front-end don't work.
- error js after drag&drop

File: Resource/template/admin/product_rank.twig [hunk 5]
```diff
@@ -19,18 +19,18 @@ You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 #}
-{% extends 'default_frame.twig' %}
+{% extends '@admin/default_frame.twig' %}
 
 {% set menus = ['product', 'class_category'] %}
```
Comment (lqdung-lockon) [inline Resource/template/admin/product_rank.twig]: please set correct navi

File: EventSubscriber.php (hunk 1 above)
Comment (ndquocphong) [reply EventSubscriber.php]: @lqdung-lockon @hoand-vn @toannguyen-lockon 
I replace EccubeEvents::FRONT_PRODUCT_INDEX_SEARCH event by using QueryCustomizer. But query customizer can not set option `wrap-queries` of paginator (see https://github.com/KnpLabs/knp-components/blob/master/doc/pager/config.md).

So I can not replace knp_pager.items by QueryCustomizer. Do you have any idea ?

File: Repository/ProductRankRepository.php (hunk 2 above)
Comment (ndquocphong) [reply Repository/ProductRankRepository.php:37]: @lqdung-lockon @toannguyen-lockon @hoand-vn 
When extend `AbstractRepository`, I must define entity which use for repository. But ProductRank does not have any entity.

File: Resource/template/admin/product_rank.twig (hunk 5 above)
Comment (ndquocphong) [reply Resource/template/admin/product_rank.twig]: @lqdung-lockon @toannguyen-lockon @hoand-vn done

File: Resource/template/admin/product_rank.twig (hunk 4 above)
Comment (ndquocphong) [reply Resource/template/admin/product_rank.twig]: @lqdung-lockon @toannguyen-lockon @hoand-vn done

**Comment by hoand-vn** [comment]
finish review

**Pull request closed without merging**